test/functional/wallet_conflicts.py

```diff
@@ -123,5 +130,165 @@ def run_test(self):
         assert_equal(former_conflicted["confirmations"], 1)
         assert_equal(former_conflicted["blockheight"], 217)
 
+    def test_mempool_conflict(self):
```

Comment:
Not sure if in scope for this PR, but it would be nice to have a test for how the wallet sees descendants of conflicting transactions (e.g. if Bob creates a child off of tx1 or tx2).

Reply:
Yes, I will add these in a follow-up PR.

Reply:
Actually, I have just added a test in this PR.